service/src/test/java/io/camunda/service/license/CamundaLicenseTest.java

```diff
@@ -7,27 +7,93 @@
  */
 package io.camunda.service.license;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
 
+import java.util.HashMap;
+import java.util.Map;
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 
 public class CamundaLicenseTest {
-  public static final String VALID_LICENSE = "valid license";
 
   @Test
-  public void shouldReturnTrueWhenLicenseIsValid() {
-    // given
-    final CamundaLicense camundaLicense =
-        new CamundaLicense(new MockEnvironmentVariableReaderWithValidLicense());
+  public void isValidShouldReturnTrueWhenLicenseIsValid() throws InvalidLicenseException {
+    final CamundaLicense testLicense = spy(CamundaLicense.class);
+    final LicenseKey mockKey = mock(LicenseKey.class);
 
-    assertTrue(camundaLicense.isValid());
+    Mockito.doReturn(mockKey).when(testLicense).getLicenseKey(anyString());
+
+    testLicense.initializeWithLicense("whatever");
+
+    assertTrue(testLicense.isValid());
+  }
+
+  @Test
+  public void isValidShouldReturnFalseWhenLicenseIsInvalid() throws InvalidLicenseException {
+    final CamundaLicense testLicense = spy(CamundaLicense.class);
+
+    Mockito.doThrow(new InvalidLicenseException("test exception"))
+        .when(testLicense)
+        .getLicenseKey(anyString());
+
+    testLicense.initializeWithLicense("whatever");
+
+    assertFalse(testLicense.isValid());
   }
```

Comment:
```suggestion
  @Test
  public void shouldReturnFalseWhenLicenseIsInvalid() throws InvalidLicenseException {
    // given
    final CamundaLicense testLicense = spy(CamundaLicense.class);

    Mockito.doThrow(new InvalidLicenseException("test exception"))
        .when(testLicense)
        .getLicenseKey(anyString());

    // when
    testLicense.initializeWithLicense("whatever");
    
    // then
    assertFalse(testLicense.isValid());
  }
```

Reply:
We normally have a clearer separation of `given, when, then` via comments it helps the reader to better understand such tests.